app/assets/javascripts/c_rater.coffee

```diff
@@ -0,0 +1,141 @@
+class ArgumentationBlockController
+  QUESTION_SEL = '.question'
+  QUESTION_FROMS_SEL = '.arg-block ' + QUESTION_SEL + ' form'
+  SUBMIT_BTN_SEL = '#ab-submit'
+  FEEDBACK_SEL = '.ab-feedback'
+  FEEDBACK_ID_SEL = '#feedback_on_answer_'
+  FEEDBACK_TEXT_SEL = '.ab-feedback-text'
+  DIRTY_MSG_SEL = '.ab-dirty'
+  FEEDBACK_HEADER_SEL = '.ab-feedback-header'
+
+  constructor: ->
+    @$submitBtn = $(SUBMIT_BTN_SEL)
+    @question = {}
+    for q in $(QUESTION_FROMS_SEL)
+      isFeedbackDirty = $(q).closest(QUESTION_SEL).find(FEEDBACK_SEL).data('dirty')
+      @question[q.id] = {
+        # It will be updated by answer_for or no_answer_for event handler.
+        answered: false,
+        dirty: isFeedbackDirty,
+        # 'dirty-data' ensures that question will be always considered as dirty unless it's submitted.
+        data: if isFeedbackDirty then 'dirty-data' else $(q).serialize(),
+        formElement: q,
+        dirtyMsgElement: $(q).closest(QUESTION_SEL).find(FEEDBACK_SEL).find(DIRTY_MSG_SEL)[0]
+      }
+    @registerListeners()
+
+  registerListeners: ->
+    $(document).on 'answer_for', (e, opt) =>
+      @updateQuestion(opt.source, true)
+      @updateView()
+    $(document).on 'no_answer_for', (e, opt) =>
+      @updateQuestion(opt.source, false)
+      @updateView()
+
+    @$submitBtn.on 'click', (e) =>
+      @submitButtonClicked(e)
+
+  updateQuestion: (id, answered) ->
+    q = @question[id]
+    # Undefined means that this question isn't part of the argumentation block.
+    return unless q
+    q.answered = answered
+    q.dirty = q.data != $(q.formElement).serialize()
+
+  submitButtonClicked: (e) ->
+    unless @allQuestionAnswered()
+      return modalDialog(false, 'Please answer all questions in the argumentation block.')
```

Comment:
The user visible strings in this file should be moved into the en.yml file.  I'm not sure how this is done in javascript files, but I'm pretty sure there is example of this in the LARA code somewhere.